GDX: Generic Poi3DLayer and GdxModelLayer

Gustl22 opened the pull request:
Unfortunately gdx assets (trees) aren't found in android examples.. so I copied them in the assets folder of android examples.
I didn't work with assets so far, maybe you can correct that.

devemux86 commented:
Such assets usually belong to the apps, not in the library, so should exist in vtm-android-example and vtm-playground.

devemux86 commented:
Constructor can have the `Tag` and the loaded `Model` and need to handle / change the `treeA_root` somehow.

That way users can use conveniently the layer for custom models. Isn't that its purpose as library layer?

devemux86 commented:
`treeA.g3dj` can move in vtm-themes where seems to belong.

Gustl22 commented:
Now the layers are more generic. I added a default configuration for most relevant tags. Of course the layers are customizable. Loading assets suddenly works in Android and Desktop...

I wonder, how we can use these layers in regular Mapsforge map activities. But there the asset path is not found. (See code...)

I don't see an advance of `SharedModel` compared to the `ModelInstance` of Gdx, so I removed it (?).

devemux86 commented:
Thanks for the work!

But we try always to do multiple things in the simplest way step by step. :slightly_smiling_face: 
A PR with simultaneous package changes and classes modifications cannot be used because it does not provide a clear history for the future and is most difficult to review.

All those things should to be split / merged in absolute discrete steps.
Plus first step is to have current functionality working again in the new structure and then add separately new features or models on it.

So we need to go back and either work directly on vtm-playground or move classes and then work on them.

Either way please better discuss first any changes.
As my previous very fast check of the code showed that only "minor" things could be required for the expected result, without so much core changes(?).

devemux86 commented on vtm-themes/src/org/oscim/theme/VtmModels.java:
```diff
@@ -0,0 +1,51 @@
+    TREE_OAK("models/natural/tree_oak.g3db"),
+    TREE_FIR("models/natural/tree_fir.g3db"),
+    TREE("models/natural/treeA.g3dj"),
+    TEST("models/test/test.g3db");
```
Models should exist in non-binary (or also in non-binary) form so that users can easily adapt them to their liking.

Gustl22 replied:
> All those things should to be split / merged in absolute discrete steps.

I absolutely support that. But I wanted to show a complete overview of what is done, as you may have better proposals for the overall structure (as you had).

Core changes are needed to provide an all-embracing workflow and I don't want to start with emergency solutions. It also depends on what the expected result should look like.

My next steps would be:
- Remove `SharedModel`
- Extend AssetAdapter
- Add models to assets
- Improve code in Playground
- Port stuff to gdx module

Thanks for reviewing.

Gustl22 commented:
> All those things should to be split / merged in absolute discrete steps.

I absolutely support that. But I wanted to show a complete overview of what is done, as you may have better proposals for the overall structure (as you had).

Core changes are needed to provide an all-embracing workflow and I don't want to start with emergency solutions. It also depends on what the expected result should look like.

My next steps would be:
- Remove `SharedModel`
- Extend AssetAdapter
- Add models to assets
- Improve code in Playground
- Port stuff to gdx module

Thanks for reviewing.

devemux86 commented:
Thanks, that seems a good work flow.

devemux86 commented on vtm-android/src/org/oscim/android/AndroidAssets.java:
```diff
@@ -35,6 +35,13 @@ private AndroidAssets(Context ctx) {
+        // TODO get android asset path
+        throw new UnsupportedOperationException("Get path of Android assets currently not supported");
+        // return "file:///android_asset/" + file;
+    }
```
Can models work with `InputStream` like all our current assets?
Then without changes the `AssetAdapter` is ready to read them.

Gustl22 replied:
I don't think so. [There](https://github.com/libgdx/libgdx/wiki/Managing-your-assets)'s no mention of InputStream. 

I decided to let it handle GDX as it's the only affected platform (see new commit).

devemux86 replied:
That's unfortunate.

BTW can close this large PR and focus on next smaller ones per feature.
